site/src/Librecores/ProjectRepoBundle/Repository/CommitRepository.php

```diff
@@ -351,6 +351,120 @@ private function getCommitHistogramByYear(
             ->getQuery()
             ->getResult('group');
 
+        return $this->zeroFillMissingYears($result, $start, $end);
+    }
+
+    /**
+     * Get a histogram of contributors over a range of dates
+     *
+     * @param SourceRepo $repo
+     * @param int $bucket one of the constants 'INTERVAL_DAY', 'INTERVAL_WEEK'
+     *                    'INTERVAL_MONTH', 'INTERVAL_YEAR', which specifies
+     *                     the histogram bucket size
+     * @param \DateTimeImmutable $start start date of commits
+     * @param \DateTimeImmutable $end end date of commits
+     * @return array associative array of a time span index and commits in that
+     *               time span
+     */
+    public function getCommitContributorHistogram(
+        SourceRepo $repo,
+        int $bucket,
+        \DateTimeImmutable $start = null,
+        \DateTimeImmutable $end = null
+    ): array
+    {
+        // TODO: Implement other aggregations
+
+        switch ($bucket) {
+            case Dates::INTERVAL_YEAR:
+                return $this->getCommitContributorHistogramByYear($repo, $start, $end);
+            default:
+                throw new \InvalidArgumentException(
+                    "Invaid value $bucket for \$bucket"
+                );
+        }
+    }
+
+    /**
+     * Get number of unique contributors per year
+     *
+     * @param SourceRepo $repo
+     * @param \DateTimeImmutable $start start date of commits
+     * @param \DateTimeImmutable $end end date of commits
+     *
+     * @return array associative array of a time span index and commits in that
+     *               time span
+     */
+    private function getCommitContributorHistogramByYear(
```

Comment:
This function only returns contributions happening through commits I assume? Please make this a bit more clear in the phpdoc, "CommitContributors" is hard to understand otherwise.